src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -665,6 +665,10 @@ public boolean isFinal() {
         return (tsym.flags() & FINAL) != 0;
     }
 
+    public boolean isValueBased() {
+        return (tsym.flags_field & VALUE_BASED) != 0;
```

Comment:
We have encountered a NPE in javac:
```
  java.lang.NullPointerException: Cannot read field "flags_field" because "this.tsym" is null
  	at jdk.compiler/com.sun.tools.javac.code.Type.isValueBased(Type.java:669)
  	at jdk.compiler/com.sun.tools.javac.comp.Check.checkRequiresIdentity(Check.java:5739)
  	at jdk.compiler/com.sun.tools.javac.comp.Attr.visitNewClass(Attr.java:2918)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1915)
...
```
It looks like `tsym` can be null https://github.com/openjdk/jdk/blob/60f1f53b18d065f110936e71b7430d7e365881d4/src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java#L517-L519

Reply:
thanks for the report I will update the PR